mill-scalafix/src/com/goyeau/mill/scalafix/ScalafixModule.scala

```diff
@@ -25,7 +25,7 @@ trait ScalafixModule extends ScalaModule {
         T.ctx().log,
         repositoriesTask(),
         filesToFix(sources()).map(_.path),
-        Seq(semanticDbData().path),
+        classpath = (compileClasspath() ++ localClasspath() ++ Seq(semanticDbData())).iterator.toSeq.map(_.path),
```

Comment:
A potential further improvement to reduce the added overhead would be to avoid triggering any compilation if none of the rules is semantic (`rulesThatWillRun.exists(_.kind.isSemantic)`) like it's done in sbt-scalafix. Not sure how/if this (dynamic tasks in sbt) can be achieved in Mill.

Reply:
Could be tricky, as the task graph can't be changed once it has started. But we have some options at construction time, depending on how lightweight (fast) a check can be.

Anyways, this should be done in a separate PR.